### PR TITLE
[wasm][debug-proxy] Fix crash and send back failed message.

### PR DESCRIPTION
**Pull request by kjpou1**
When the proxy runs into an error parsing properties request it will crash and end the debugging session unexpectedly.  Instead log the error to stdout right now and send back a failed message.

The locals window in VSC returns empty.  This may be what is happening as reported by Blazor but not sure if it is the exact error they reported.

Also, updates `Mono.Cecil` and `Newtonsoft` to current versions.

### CODE REVIEW

**Comment by kjpou1** [comment]
@monojenkins build Linux WebAssembly

**Comment by kjpou1** [comment]
@monojenkins build Linux WebAssembly

**Comment by kjpou1** [comment]
@monojenkins build Linux WebAssembly

File: sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs [hunk 1]
```diff
@@ -427,34 +427,36 @@ async Task GetDetails(int msg_id, int object_id, CancellationToken token, string
...
-				if (fieldName.Contains("k__BackingField")){
-					fieldName = fieldName.Replace("k__BackingField", "");
-					fieldName = fieldName.Replace("<", "");
-					fieldName = fieldName.Replace(">", "");
-				}
-				var value = values [i + 1] ["value"];
-				if (((string)value ["description"]) == null)
-					value ["description"] = value ["value"]?.ToString ();
+				var values = res.Value?["result"]?["value"]?.Values<JObject>().ToArray() ?? Array.Empty<JObject>();
+				foreach (var valss in values)
```
Comment (lewing) [inline sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs]: either use Debug or remove

File: sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs [hunk 2]
```diff
@@ -486,46 +488,52 @@ async Task GetScopeProperties (int msg_id, int scope_id, CancellationToken token
...
+				var values = res.Value? ["result"]? ["value"]?.Values<JObject> ().ToArray ();
+
+				var var_list = new List<JObject> ();
+				int i = 0;
+				// Trying to inspect the stack frame for DotNetDispatcher::InvokeSynchronously
+				// results in a "Memory access out of bounds", causing 'values' to be null,
+				// so skip returning variable values in that case.
+				while (values != null && i < vars.Length && i < values.Length) {
+					var value = values [i] ["value"];
+					Console.WriteLine ($"Value: {value}");
```
Comment (lewing) [inline sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs]: either use Debug or remove

**Pull request merged**